### ISSUE TITLE
Verifying a remote user didn't happen in DM

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#13204]
* `@matthewtest401:matrix.org` verified `@matthew:arasphere.net`
 * The verif succeeded
 * The DM didn't contain an audit trail.
 * matthewtest401 then saw arasphere as green shield
 * arasphere saw matthewtest401 as black shield :-/

**Comment by bwindels** [comment]
I imagine it was falling back to legacy user verification because it could not find the cross-signing keys for the user it was going to verify.

**Comment by bwindels** [comment]
The logs from `@matthew:arasphere.net` indeed show this right before starting the verification request:
```
2020-04-16T12:33:49.346Z I got cross-signing keys for @matthewtest401:matrix.org: {}
```

**Comment by bwindels** [comment]
When quering keys now (from matrix.org) for `@matthewtest401:matrix.org` I get a master key of `dn699wr2FITbozlGId4YH8vsb+BlhYC/AuJzo34/aCs`.

**Comment by ara4n** [comment]
> When expanding the session section in user info @ara4n, was the device you verified marked green at least?

Yes:

<img width="267" alt="Screenshot 2020-04-16 at 15 51 41" src="https://user-images.githubusercontent.com/1294269/79471184-2fe04580-7ffa-11ea-9ee5-037f7339b4a6.png">

**Comment by bwindels** [comment]
This feels related to https://github.com/vector-im/riot-web/issues/12973 and https://github.com/matrix-org/synapse/issues/7276 then.

**Comment by ara4n** [comment]
see also https://github.com/vector-im/riot-web/issues/13248

**Comment by Thatoo** [comment]
I experiment that problem too as I was explaining in here : https://github.com/vector-im/riot-web/issues/13566

My shield is green on his devices but his shield keep being black on my devices (both Riot-desktop and RiotX). I have all his devices verified though (legacy way)... but I can still start a cross-signing verification with him from Riot-desktop (he can't) that won't lead to a better situation. It will say that it succeeded but his shield will remain black.

**Comment by bwindels** [comment]
@Thatoo can you please file a separate issue and submit your debug logs from "help & about" in the settings?

**Comment by Thatoo** [comment]
I have done it in here : https://github.com/vector-im/riot-web/issues/13824
Thank you for your help.

**Comment by ShadowJonathan** [comment]
Can't reproduce, this probably needs a very specific homeserver setup.

This also feels like a cascade of client and server bugs, specifically synapse ones.

**Comment by kittykat** [comment]
I'm going to close this issue for now as there is no futher information that we could request for debugging at this point due to the age of the issue, we haven't had any more recent reports and the code has changed since it was filed. Please re-open if you are still experiencing it.
